go2rtc_client/rest.py

```diff
@@ -144,10 +144,15 @@ def __init__(self, websession: ClientSession, server_url: str) -> None:
         self.streams: Final = _StreamClient(self._client)
         self.webrtc: Final = _WebRTCClient(self._client)
 
+    @classmethod
     @handle_error
-    async def validate_server_version(self) -> None:
+    async def validate_server_version(
+        cls, websession: ClientSession, server_url: str
+    ) -> None:
```

Comment:
Is this really more convenient? Won't this just mean we have to pass the session and url twice to two calls instead of passing them once to the first of two calls, when validating the server version and creating a client? We also just save two instance creation calls compared to creating the client from the beginning.

Reply:
I had the same discussion with Robert. From the *user's* point of view, I think it makes sense to make it a class method; there's no need for a client if the server has the wrong version.
From the current implementation point of view it's as you say currently not useful since the implementation creates the client objects.
At the same time, the client objects are very lightweight and cost very little to create.